server/container_update_resources.go

```diff
@@ -57,23 +62,18 @@ func toOCIResources(r *pb.LinuxContainerResources) *rspec.LinuxResources {
 	}
 }
 
-func (s *Server) updateContainerResources(ctx context.Context, c *oci.Container, resources *rspec.LinuxResources) (retErr error) {
+func (s *Server) updateContainerResources(c *oci.Container, resources *rspec.LinuxResources) error {
+	// generate updated spec under the <bundlePath>/config.updated.json
 	if err := updateContainerSpec(c, resources); err != nil {
 		return err
 	}
 
-	defer func() {
-		if retErr != nil {
-			oldSpec := c.Spec()
-			oldSpecGen := generate.Generator{Config: &oldSpec}
-			saveOptions := generate.ExportOptions{}
-			if err := oldSpecGen.SaveToFile(filepath.Join(c.BundlePath(), "config.json"), saveOptions); err != nil {
-				log.Warnf(ctx, "failed to update container %q spec", c.ID())
-			}
-		}
-	}()
-
-	return s.Runtime().UpdateContainer(c, resources)
+	if err := s.Runtime().UpdateContainer(c, resources); err != nil {
+		return err
+	}
+
+	// replace the old config file with the new one
+	return os.Rename(filepath.Join(c.BundlePath(), configUpdatedFile), filepath.Join(c.BundlePath(), configFile))
```

Comment:
should the update fail, do we want to remove the now-stale `config.updated.json`?

Reply:
yeah I think that would be good